Horny curse revamp

Stadler76 opened the pull request:
This is the first version of a revamped Horny curse with choices for the preferred flavor for the player.

Stadler76 committed:
First version of an revamped Horny curse

emmeken commented on src/curses.twee:
```diff
@@ -1,5 +1,13 @@
 :: Init Curses
 
+<<set $curseData = {
+	77: {
+		variation: "",
+		adjective: "",
+		count: 0
+	}
+}>>
```
I don't think there is a real advantage to having this as a hierarchical variable. Even if there will be more curses which require variations to be stored, it's unlikely that there will be enough similarity with this curse to be able to use generic code for it.

```suggestion
<<set $hornVariation = "">>
<<set $hornAdjective = "">>
<<set $hornCount = 0>>
```

Stadler76 replied:
I'm currwently applying that manually. Just as a side note: I'll leave that suggested chunk in `curses.twee` as I think, that it fits there better. Maybe later split that up into Init Curses and Init Curse Data?

emmeken commented on src/global.twee:
```diff
@@ -1672,7 +1680,16 @@ How many dubloons would you like to pay back?
 	<<if $tempCurse.name == "Horny">>
-		You always wondered why all the depictions of incubi and succubi that you'd seen just has them standing around looking pretty or using magic, or fighting like humans with extra agility. They always have those horns on their heads, but they never seem to use them like horns are supposed to be used, for goring or ramming. Well, now that you have your own thanks to the Horny Curse, you understand. ...
+		<<switch $curseData[77].variation>>
+			<<case "demonic">>
+				You always wondered why all the depictions of incubi and succubi that you'd seen just has them standing around looking pretty or using magic, or fighting like humans with extra agility. They always have those horns on their heads, but they never seem to use them like horns are supposed to be used, for goring or ramming. Well, now that you have your own thanks to the Horny Curse, you understan ...
+			<<case "draconic">>
+				You always wondered why all the depictions of fearsome dragons that you'd seen just has them standing around looking pretty or using magic, or fighting like humans with extra agility. They always have those horns on their heads, but they never seem to use them like horns are supposed to be used, for goring or ramming. Well, now that you have your own thanks to the Horny Curse, you understand. ...
+			<<case "unicorn">>
+				You always wondered why all the depictions of beautiful unicorns that you'd seen just has them standing around looking pretty or using magic, or fighting like humans with extra agility. They always have those horns on their heads, but they never seem to use them like horns are supposed to be used, for goring or ramming. Well, now that you have your own thanks to the Horny Curse, you understan ...
+			<<case "rhino">>
+				You always wondered why all the depictions of rhinos that you'd seen just has them standing around looking pretty or using magic, or fighting like humans with extra agility. They always have those horns on their heads, but they never seem to use them like horns are supposed to be used, for goring or ramming. Well, now that you have your own thanks to the Horny Curse, you understand. They're n ...
```
Are these descriptions going to be customized further? If not, I think it would be better to move the identical part out of the `switch` macro, to avoid duplication.

Stadler76 replied:
Yeah, they should be customized further.

emmeken commented on src/layer6.twee:
```diff
@@ -204,7 +204,7 @@ If you decide to start foraging for food or water, it means that you will not re
 	<</if>>
 <</if>>
-<<if setup.takenCurseCount("Horny") >= 2>>
+<<if $StoredCurse.name == "Horny" || $curseData[77].count >= 1>>
```
```suggestion
<<if setup.takenCurseCount("Horny") >= 1>
```
Continuing the use of `takenCurseCount()` will make things easier when generalizing this code later, for example by comparing the current count to a per-curse `maxCount`.

emmeken commented on src/layer6.twee:
```diff
@@ -761,6 +761,40 @@ You have successfully taken on the $curse77.name curse. Hopefully you are ok wit
 <</if>>
 
+:: Horny Customization
+What type of horn(s) would you like to gain?
+
+<<radiobutton "$curseData[77].variation" "demonic" checked>> Demon
+<<radiobutton "$curseData[77].variation" "draconic">> Dragon
+<<radiobutton "$curseData[77].variation" "unicorn">> Unicorn
+<<radiobutton "$curseData[77].variation" "rhino">> Rhino
+
+How many horns would you like to gain?
+
+<<radiobutton "$curseData[77].count" 1 checked>> One <<radiobutton "$curseData[77].count" 2 >> Two
```
It might make more sense to have the number of horns be implied by the type. For example, ["unicorn" literally means "one horn"](https://www.etymonline.com/word/unicorn).

Is demon horn a single type? I've seen demon horns drawn as smooth cow-like horns, as ribbed goat-like horns and as large curved sheep-like horns.

Stadler76 replied:
I started off with only 1 unicorn horn but then FloricSpacer said "I mean, I don’t see why someone couldn’t get 2 unicorn style horns if they wanted?"

I could add more demon horn types although this could be done with the adjectives.

emmeken commented on src/layer6.twee:
```diff
@@ -761,6 +761,40 @@ You have successfully taken on the $curse77.name curse. Hopefully you are ok wit
+
+<<radiobutton "$curseData[77].count" 1 checked>> One <<radiobutton "$curseData[77].count" 2 >> Two
+
+Enter one or more descriptive adjectives (optional)
```
Maybe add some kind of example here, so players know how this will be used.

Stadler76 replied:
Here's the examples I came up with so far:
![grafik](https://user-images.githubusercontent.com/13157984/177058728-c17274bd-2272-4156-851f-a397397c2dd7.png)

That should (at least for now) resolve the above, too.

emmeken commented on src/layer6.twee:
```diff
@@ -871,7 +905,7 @@ You have successfully taken on the $curse84.name curse. Hopefully you are ok wit
 		<<break>>
 	<</if>>
-	<<if ($temp1 == 76) && setup.takenCurseCount("Horny") < 2>>
+	<<if ($temp1 == 76) && $StoredCurse.name != "Horny" && $curseData[77].count == 0>>
```
```suggestion
	<<if ($temp1 == 76) && setup.takenCurseCount("Horny") == 0>
```

emmeken commented on src/widgets.twee:
```diff
@@ -1087,3 +1087,15 @@
+<<widget "HornGrowthDesc">>
+	<<if _args[1] == 1>>a<<else>>two<</if>>
+	<<switch _args[0]>>
+		<<case "demonic" "draconic">>
+			_args[0]-looking\
+		<<case "unicorn" "rhino">>
+			_args[0]-like\
+	<</switch>>
+	horn<<if _args[1] > 1>>s<</if>>\
```
Do you need the backslashes when the widget itself is tagged as `nobr`?

Stadler76 replied:
Only the last one is needed, or there would be an unwanted space added after horn(s).
Resolved this by removing the two backslashes after -looking and -like.

emmeken reviewed (changes requested):
I think you're on the right track, but some things could be changed to make future maintenance easier.

As for the exact customization to be offered to the player, this is always going to be subjective, but I think it might be useful to discuss some options before picking one.

Stadler76 committed:
Apply suggestions from code review

`setup.takingCurseCount` still works as intended there.

Co-authored-by: emmeken <emmeken@ymail.com>

Stadler76 committed:
Merge branch 'FloricSpacer-preview' into horny-upgraded

Stadler76 committed:
Resolved emmekens reviews.

Stadler76 committed:
Updated the horny random event texts as per Nerevars suggestion

Thanks go to Nerevar to help with updating the texts.
I've added the widgets `<<Numerical>>` and `<<PluralS>>` to avoid littering it with `<<if $hornCount == 1>>...<<else>>...<</if>`

Stadler76 committed:
Merge branch 'FloricSpacer-preview' into horny-upgraded

Stadler76 committed:
Merge branch 'FloricSpacer-preview' into horny-upgraded

Ilyza91 commented:
I looked at it, and by itself itself it works very nicely. Maybe you can clearly indicate the double corruption for two horns somewhere. However, it still does have it's problems when combined with managed misfortune and I imagine other mechanics to reverse curses, maybe you can look at that as well.

Stadler76 committed:
Fixed storing Horny in Managed Misfortune.
